services/src/test/scala/cromwell/pipeline/service/ProjectServiceTestImpl.scala

```diff
@@ -0,0 +1,53 @@
+package cromwell.pipeline.service
+import cromwell.pipeline.datastorage.dto._
+import cromwell.pipeline.model.wrapper.UserId
+
+import scala.concurrent.Future
+
+class ProjectServiceTestImpl(projects: Seq[Project], testMode: TestMode) extends ProjectService {
+
+  private[service] def getUserProjectById(projectId: ProjectId, userId: UserId): Future[Project] =
+    testMode match {
+      case WithException(exc) => Future.failed(exc)
+      case _                  => Future.successful(projects.head)
+    }
+
+  def getUserProjectByName(namePattern: String, userId: UserId): Future[Project] =
+    testMode match {
+      case WithException(exc) => Future.failed(exc)
+      case _                  => Future.successful(projects.head)
+    }
+
+  def addProject(request: ProjectAdditionRequest, userId: UserId): Future[Either[VersioningException, Project]] =
+    testMode match {
+      case WithException(exc: VersioningException) => Future.successful(Left(exc))
+      case WithException(exc)                      => Future.failed(exc)
+      case _                                       => Future.successful(Right(projects.head))
+    }
+
+  def deactivateProjectById(projectId: ProjectId, userId: UserId): Future[Project] =
+    testMode match {
+      case WithException(exc) => Future.failed(exc)
+      case _                  => Future.successful(projects.head)
+    }
+
+  def updateProjectName(projectId: ProjectId, request: ProjectUpdateNameRequest, userId: UserId): Future[ProjectId] =
+    testMode match {
+      case WithException(exc) => Future.failed(exc)
+      case _                  => Future.successful(projects.head.projectId)
```

Comment:
Almost the same problem as with `UserServiceTestImpl.addUser` or `RunServiceTestImpl. addRun`.
Either return a failed future or just return `projectId` that you _already_ have

Reply:
Fixed